### DISCUSSION TITLE
Retrieving the publication date of a YouTube video

### DISCUSSION THREAD

**Discussion by kenNash** [johnlindquist/kit#1507] [Q&A]
Hi Kiters!

I'm trying to retrieve the publication date of a YouTube video, as a part of a script.

I've found some code that uses the `DOMParser()` function, but Kit says it doesn't recognise it.
Not an expert here. A learner, instead.

Any idea (not necessarily related to DOMParser, for me the job done is enough ;-))?

Thanks in advance

**Comment by johnlindquist** [comment]
Do you have a script that you can share? I just don't understand how you're getting the data from YouTube. Are you using a YouTube API or are you scraping YouTube? I'm just curious how you're getting the video. Because I don't think that using that DOM parser function would be necessary.

**Comment by kenNash** [comment]
Sure!

The purpose of the script is the following: starting from a Safari page where a YouTube video is shown, to generate an Obsidian note (a markdown file) containing some info about it. What I still have to do is to append `(2024)` (for a video uploaded this year) to the name of the note.

Here's how my script looks like at the moment (without the year part):
```
// Name: Youtube Video to Obsidian Note
// Description: Youtube Video to Obsidian Note
// Author: kenNash

import "@johnlindquist/kit";

let jxa = await npm("@jxa/run")

const VAULT_NAME = "Documents/Obsidian/"

let page = await jxa.run(
  () => {
    let windows = Application("Safari").windows()
    let tab = windows[0].currentTab()

    return {
      url: tab.url(),
      title: tab.name()
    }
  }
)

// await div(page.url)
// await div(page.title)

page.title = page.title
.replace(" - YouTube", "")
.replaceAll(":"," -")
.replaceAll("/",">")
.replaceAll("|","@")
// await div(page.title)

let noteFolder = home(VAULT_NAME, "Resources", "Videos")
noteFolder = noteFolder + "/" + page.title
// await div(noteFolder)

let exists = await pathExists(`"${noteFolder}"`)
if (!exists) {
  await exec(`mkdir -p "${noteFolder}"`)
}

let noteTitle = "🎬 " + page.title
// await div(noteTitle)

let notePath = noteFolder + "/" + noteTitle + ".md"
// await div(notePath)

let noteContent = 
"---\n"
+ "type: Video\n"
+ "---\n---\n\n"
+ "> [!info]- HEADER\n> \n"
+ "> `Title`:: " + page.title + "\n> \n"
+ "> \`Author\`:: " + "\n> \n"
+ "> \`Platform\`:: [[YouTube]]\n> \n"
+ "> `WebPage`:: " + page.url + "\n> \n"
+ "> \`Year\`:: \n> \n"
+ "> \`Part_of\`:: \n> \n"
+ "> \`Topic\`:: \n> \n"
+ "> \`Tag\`:: \n> \n"
+ "> \`Status\`:: #Status/<% tp.file.cursor(1) %>\n\n"
+ "---\n\n\n\n---\n\n"

  await writeFile(notePath, noteContent)
  await open(notePath, noteContent)
```

The code I've found that retrieves the year using `DOMParser` can be found here: https://github.com/SilentVoid13/Templater/discussions/432.

**Comment by kenNash** [comment]
And please: feel free to improve it! ;-)
